crates/rust-analyzer/src/handlers/request.rs

```diff
@@ -492,6 +492,7 @@ pub(crate) fn handle_document_diagnostics(
     if !snap.analysis.is_local_source_root(source_root)? {
         return Ok(empty_diagnostic_report());
     }
+    let source_root = snap.analysis.source_root_id(file_id)?;
```

Comment:
```suggestion
```

Reply:
wait, why this change? we need the source root in order to to call `Config::diagnostics`. Do you want me to inline this, or...?

Reply:
oh, no worries.